lib/resty/session/memcached.lua

```diff
@@ -4,26 +4,135 @@
 -- @module resty.session.memcached
 
 
-local memcached = require "resty.memcached"
-local get_name = require "resty.session.utils".get_name
+local memcached   = require "resty.memcached"
+local utils       = require "resty.session.utils"
+local buffer      = require "string.buffer"
 
 
 local setmetatable = setmetatable
-local error = error
-local null = ngx.null
+local error        = error
+local null         = ngx.null
+local time         = ngx.time
+local get_name     = utils.get_name
+local get_meta_key = utils.get_meta_key
+local get_meta_el_val = utils.get_meta_el_val
+local get_latest_valid = utils.get_latest_valid
+
+
+local function metadata_cleanup(self, memc, aud_sub_key)
+  local now     = time()
+  local retry   = 10
```

Comment:
In case cas() fails to set the value (which means it was accessed and modified after we got it via gets()), the cleanup retries 10 times before giving up. Are we ok with this? For how I see it the retry logic could also be removed.